### PR TITLE
Small fix to lattice determinization

### PR DESCRIPTION
**Pull request by vdp**
As requested by Dan in #382 

Dan, I think that in general you are trying to enforce 80 character limit, so you may find some lines a bit longish, but there are much longer lines in these files anyway, so I don't think this is a big deal (plus one could argue that in today's world, where the buying power of the YouTube fans leads to omnipresent abominations like 16:9 screens, the vertical real estate is at least as valuable).
Also, I'm not sure that the final block scopes are needed, but I wanted to make as few changes as possible.

### CODE REVIEW

File: src/fstext/determinize-lattice-inl.h [hunk 1]
```diff
@@ -547,6 +547,11 @@ template<class Weight, class IntType> class LatticeDeterminizer {
       return (state != other.state || string != other.string ||
               weight != other.weight);
     }
+    // This operator is only intended to support sorting in the function
```
Comment (danpovey) [inline src/fstext/determinize-lattice-inl.h]: Can you please implement an operator < instead?  That's the operator the sorting looks for by default, and in that case you wouldn't have to provide std::greater<Element>() to the function.  Also, this way we'll get the same order that the old code would have produced, which doesn't really matter but which is maybe less surprising.

**Comment by vdp** [comment]
@danpovey OK, I did it this way in order to reuse the already existing ">" operator in LatticeDeterminizerPruned, and added this operator LatticeDeterminizer for consistency.. Please find the corrected version (not tested yet, will recompile and test in a bit).

**Comment by danpovey** [comment]
Looks fine but I'll wait till you say it's tested.  Thanks.

**Comment by vdp** [comment]
Seems to work OK..

**Pull request merged**